Source/Engines/ModuleEngineConfigs.cs

```diff
@@ -168,6 +170,62 @@ public void UpdateTFInterops()
         }
         #endregion
 
+        #region B9PartSwitch
+        private static bool _b9psReflectionInitialized = false;
+        private static FieldInfo B9PS_moduleID;
+        private static MethodInfo B9PS_SwitchSubtype;
+        public PartModule B9PSModule;
+
+        private void InitializeB9PSIntegrationIfEnabled()
+        {
+            if (b9psModuleID != string.Empty && Utilities.B9PSFound)
+            {
+                if (!_b9psReflectionInitialized)
+                {
+                    B9PS_moduleID = Type.GetType("B9PartSwitch.CustomPartModule, B9PartSwitch")?.GetField("moduleID");
+                    B9PS_SwitchSubtype = Type.GetType("B9PartSwitch.ModuleB9PartSwitch, B9PartSwitch")?.GetMethod("SwitchSubtype");
+                    _b9psReflectionInitialized = true;
+                }
+
+                B9PSModule = GetSpecifiedModules(part, string.Empty, -1, "ModuleB9PartSwitch", false)
+                    .FirstOrDefault(m => ((string)B9PS_moduleID?.GetValue(m)).Equals(b9psModuleID));
+
+                if (B9PSModule == null)
+                    Debug.LogError($"*RFMEC* B9PartSwitch module with ID {b9psModuleID} was not found for {part}!");
+            }
+        }
+
+        private void HideB9PSVariantSelector()
+        {
+            // Hide the GUI for the `ModuleB9PartSwitch` managed by RF.
+            // This is somewhat of a hack-ish solution...
+            if (B9PSModule is PartModule module)
+            {
+                var b9psSubtypeTitle = module.Fields["currentSubtypeTitle"];
+                b9psSubtypeTitle.guiActive = false;
+                b9psSubtypeTitle.guiActiveEditor = false;
+                var b9psSubtypeSelector = module.Fields["currentSubtypeIndex"];
+                b9psSubtypeSelector.guiActive = false;
+                b9psSubtypeSelector.guiActiveEditor = false;
+                var b9psShowSelectorWindow = module.Events["ShowSubtypesWindow"];
+                b9psShowSelectorWindow.guiActive = false;
+                b9psShowSelectorWindow.guiActiveEditor = false;
+            }
+        }
+
+        public void UpdateB9PSVariant()
+        {
+            if (B9PSModule is PartModule module)
+            {
+                string subtypeName = string.Empty;
+                if (config.TryGetValue("b9psSubtypeName", ref subtypeName))
+                    B9PS_SwitchSubtype?.Invoke(module, new object[] { subtypeName });
+                else
+                    Debug.LogError($"*RFMEC* {part} has B9PS integration enabled, but the {configuration} configuration does not specify a `b9psSubtypeName`!");
```

Comment:
Is this to resolve "RO names its engine configs differently from source parts (esp BDB) that might bring Waterfall configs with them, so put the mapping here?"
Should this default to `config.name` if `config.b9psSubtypeName` is not specified?
```
string subtypeName = string.Empty;
if (!config.TryGetValue("b9psSubtypeName", ref subtypeName))
{
  Debug.LogWarning($"*RFMEC* {part} does not specify b9psSubtypeName in current config {configuration}: defaulting to {subtypeName} for B9PS switching");
  subtypeName = config.GetValue("name")
}
B9PS_SwitchSubtype?.Invoke(B9PSModule, new object[] { subtypeName });
```

Reply:
B9PS will take care of setting a default subtype if we don't set one. I'm not sure what would happen if we attempt to set a subtype name that doesn't exist (which might happen if we try to guess a default), but I'll test it.

The main purpose is because the mapping from config to subtype is not necessarily one-to-one; this is certainly the case for Waterfall plumes. Ex. all the hydrolox RL-10 configs share a single plume subtype, and the CECE-Methane config has a separate one.

But yes, the chances of RO config names matching with what stock mods use is pretty minimal.

Reply:
Turns out B9PS will error if we attempt to set a nonexistent subtype name, but it's not catastrophic (it switches back to a valid subtype):

> `[ERR 12:54:08.255] [ERROR] [Part microEngine] [ModuleB9PartSwitch 'roWaterfallMainPlumeSwitch'] Cannot assign subtype because no subtype with name = 'ModuleEnginesRF' exists`

More concerning is that `config.GetValue("name")` appears to return `"ModuleEnginesRF"`...

Reply:
My bad: config.GetValue("name") would be that -- it's the config for the ModuleEnginesRF after all!  I meant to suggest defaulting to `configuration`